### DISCUSSION TITLE
SendWhatsAppTextMessage does not send message in test project

### DISCUSSION THREAD

**Discussion by Catronis** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#17] [Q&A] [answered]
Hello, I downloaded the project a week ago and did some testing, a simple message sent correctly. I downloaded the most current version again today but it doesn't send a message, template sends normally. However, it notifies that the message has been sent, but does not reach whatsapp.

**Comment by gabrieldwight** [comment, marked as answer]
You need to reply to the template message you sent from the meta website with any text. Then the normal messages will be able to deliver to WhatsApp with the test phone number that you are using to initiate the conversation.

**Reply by gabrieldwight** [reply]
In production once your production phone number for sending WhatsApp messages is verified. The recipient does not need to respond the template message. The WhatsApp message should just deliver without any issues.

**Reply by borrmann** [reply]
@gabrieldwight oh really so once a phone number is in production I can send any message after I have sent a templated message without the user first responding to it?
Thanks for the great library btw!

**Reply by gabrieldwight** [reply]
@borrmann yes as long as your production phone number is also verified when going live with your business profile.

**Reply by borrmann** [reply]
Thanks!

**Reply by ehsanmgps** [reply]
> @borrmann yes as long as your production phone number is also verified when going live with your business profile.

Can you please explain how to verify phone number? is there any other process?

**Reply by gabrieldwight** [reply]
> > @borrmann yes as long as your production phone number is also verified when going live with your business profile.
> 
> Can you please explain how to verify phone number? is there any other process?

@ehsanmgps That's the only process for WhatsApp cloud API. You can find more information from this link [Register Phone Number](https://developers.facebook.com/docs/whatsapp/cloud-api/get-started/add-a-phone-number/)

**Comment by Catronis** [comment]
Got it, so the first message must be from a template. Thank you and congratulations on the project

**Comment by pruthvirajsinh-info** [comment]
is there any other way for getting messages without sending that one msg from template??

**Reply by gabrieldwight** [reply]
The only way is if the phone number is not a test phone number and you are using a production phone number which is verified by meta.

**Reply by theCJMan** [reply]
I found the following:
1. I have a Temp Test number from WhatsApp call this T1
2. I have my personal WhatsApp Number: call this P1

WhatsApp Code initiate a Conversation (T1):
1. You **cannot** send **any** message from T1 to P1 other than a Template message (initial Message)
2. **Until** P1 sends any message back to T1 then
3. T1 can send **any** message including tect and non template messages to P1

Client initiate Conversation (P1):
1. my P1 sends the first message to T1
2. T1 can send **any** message in reply including text, media, etc plus Templates

**Reply by gabrieldwight** [reply]
@theCJMan correct, that's the scenario for using test numbers when sending a message
